std/random.d

```diff
@@ -2198,39 +2207,38 @@ struct RandomCover(Range, UniformRNG = void)
     {
         @property size_t length()
         {
-            if (_alreadyChosen == 0)
-            {
-                return _input.length;
-            }
+            if (_isEmpty)
+                return 0;
             else
-            {
-                return (1 + _input.length) - _alreadyChosen;
-            }
+                return _input.length - _alreadyChosen;
         }
     }
 
     @property auto ref front()
     {
-        if (_alreadyChosen == 0)
-        {
-            popFront();
-        }
+        import std.exception : enforce;
+
+        enforce(!_isEmpty, "Cannot use front on an empty range");
         return _input[_current];
     }
 
     void popFront()
     {
-        if (_alreadyChosen >= _input.length)
+        if (_isEmpty)
+            return;
```

Comment:
FWIW this is undefined behavior - the range API explicitely forbids this, so the check is unnecessary:

> r.popFront can be called if and only if evaluating r.empty has, or would have, equaled false.

http://dlang.org/phobos/std_range_primitives.html